iceberg/iceberg-catalog/src/main/java/org/apache/iceberg/hive/client/HiveRESTCatalogClient.java

```diff
@@ -59,44 +56,58 @@
 import org.apache.iceberg.catalog.Namespace;
 import org.apache.iceberg.catalog.TableIdentifier;
 import org.apache.iceberg.exceptions.NoSuchTableException;
+import org.apache.iceberg.hive.HMSTablePropertyHelper;
+import org.apache.iceberg.hive.HiveOperationsBase;
+import org.apache.iceberg.hive.HiveSchemaUtil;
+import org.apache.iceberg.hive.RuntimeMetaException;
 import org.apache.iceberg.relocated.com.google.common.collect.Lists;
 import org.apache.iceberg.relocated.com.google.common.collect.Maps;
 import org.apache.iceberg.rest.RESTCatalog;
 import org.apache.thrift.TException;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-public class HiveIcebergRESTCatalogClientAdapter extends BaseMetaStoreClient {
+public class HiveRESTCatalogClient extends BaseMetaStoreClient {
 
-  private static final Logger LOG = LoggerFactory.getLogger(HiveIcebergRESTCatalogClientAdapter.class);
   public static final String NAMESPACE_SEPARATOR = ".";
   public static final String NAME = "name";
   public static final String LOCATION = "location";
   public static final String ICEBERG_CATALOG_TYPE = "iceberg.catalog.default_iceberg.type";
   public static final String DB_OWNER = "owner";
   public static final String DB_OWNER_TYPE = "ownerType";
   public static final String DEFAULT_INPUT_FORMAT_CLASS = "org.apache.iceberg.mr.hive.HiveIcebergInputFormat";
-  public static final String DEFAULT_OUTPUT_FORMAT_CLASS
-      = "org.apache.iceberg.mr.hive.HiveIcebergOutputFormat";
+  public static final String DEFAULT_OUTPUT_FORMAT_CLASS = "org.apache.iceberg.mr.hive.HiveIcebergOutputFormat";
   public static final String DEFAULT_SERDE_CLASS = "org.apache.iceberg.mr.hive.HiveIcebergSerDe";
-  public static final String CATALOG_CONFIG_PREFIX = "iceberg.rest-catalog.";
   public static final String WAREHOUSE = "warehouse";
+
+  private static final Logger LOG = LoggerFactory.getLogger(HiveRESTCatalogClient.class);
   private final Configuration conf;
   private RESTCatalog restCatalog;
-
   private final long maxHiveTablePropertySize;
 
-  public HiveIcebergRESTCatalogClientAdapter(Configuration conf) {
+  public HiveRESTCatalogClient(Configuration conf, boolean allowEmbedded) {
```

Comment:
where do you use `allowEmbedded`?

Reply:
`HiveMetaStoreClientBuilder` instantiates `HiveRESTCatalogClient` via reflection and it expects constructor with this parameter:

```
      baseMetaStoreClient = JavaUtils.newInstance(mscClass,
          new Class[]{Configuration.class, boolean.class},
          new Object[]{conf, allowEmbedded});
```